src/reducers/computeRenderedColumns.js

```diff
@@ -94,47 +97,120 @@ function calculateRenderedColumnRange(state, columnAnchor) {
-  const firstBufferIdy = Math.max(firstViewportIdy - bufferColumnCount, 0);
-  for (columnIdy = firstBufferIdy; columnIdy < firstViewportIdy; columnIdy++) {
-    updateColumnWidth(state, columnIdy);
+  let firstViewportCol = Math.min(startIdy, endIdy);
+  const firstBufferCol = Math.max(firstViewportCol - bufferColumnCount, 0);
+  for (columnIdx = firstBufferCol; columnIdx < firstViewportCol; columnIdx++) {
+    updateColumnWidth(state, columnIdx);
   }
 
   // Loop to walk the trailing buffer
-  const endViewportIdy = Math.max(startIdy, endIdy) + 1;
-  const endBufferIdy = Math.min(endViewportIdy + bufferColumnCount, columnCount);
-  for (columnIdy = endViewportIdy; columnIdy < endBufferIdy; columnIdy++) {
-    updateColumnWidth(state, columnIdy);
+  const endViewportCol = Math.max(startIdy, endIdy) + 1;
+  const endBufferCol = Math.min(endViewportCol + bufferColumnCount, columnCount);
+  for (columnIdx = endViewportCol; columnIdx < endBufferCol; columnIdx++) {
+    updateColumnWidth(state, columnIdx);
   }
 
-  // Calculate offset needed to position column row at the end of viewport
+  // Calculate offset needed to position column at the end of viewport
   // This should be negative and represent how far the first column needs to be offscreen
   if (lastIndex !== undefined) {
     firstOffset = Math.min(availableScrollWidth - totalWidth, 0);
   }
 
-  state.firstColumnIndex = firstViewportIdy;
-  state.endColumnIndex = endViewportIdy;
+  state.firstColumnIndex = firstViewportCol;
+  state.endColumnIndex = endViewportCol;
   state.firstColumnOffset = firstOffset;
 
   return {
-    endBufferIdy,
-    endViewportIdy,
-    firstBufferIdy,
-    firstViewportIdy,
+    endBufferCol,
+    endViewportCol,
+    firstBufferCol,
+    firstViewportCol,
   };
 }
 
+/**
+ * Determine the range of column groups to render (buffer and viewport)
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   firstIndex: number,
+ *   firstOffset: number,
+ *   lastIndex: number,
+ * }} columnAnchor
+ * @param {{
+ *   endBufferCol: number,
+ *   endViewportCol: number,
+ *   firstBufferCol: number,
+ *   firstViewportCol: number,
+ * }} columnRange
+ * @private
+ */
+function calculateRenderedColumnGroups(state, columnAnchor, columnRange) {
+  const bufferColumnCount = 0; // TODO (pradeep): calculate this similar to bufferRowCount
+  const { columnGroupBufferSet, columnGroupOffsetIntervalTree } = state;
+
+  const { availableScrollWidth, scrollableColumns, scrollableColumnGroups, columnGroupProps, columnGroupIndex } = columnWidths(state);
+  const columnCount = scrollableColumns.length;
+  const columnGroupCount = scrollableColumnGroups.length;
+
+  if (columnCount === 0 || columnGroupCount === 0) {
+    return;
+  }
+
+  const { firstOffset } = columnAnchor;
+  const startIdx = columnGroupIndex[scrollableColumns[Math.max(0, columnRange.firstViewportCol)].groupIdx];
```

Comment:
@wcjordan, this is worth a discussion.

To calculate the first grouped column in the viewport, we rely on firstViewportCol which is the first ungrouped column to be rendered. Similarly we also find the last column, and find the grouped column to which it belongs.

To get the grouped column index, we rely on the property `groupIdx` which each ungrouped colum has. It can be used to find out the parent index. But we can't use this index directly. Why?, because FDT allows us to specify columns freely. So freely that fixedColumns can be declared anytime before/after scrollable columns. 

This means that if we declared 5 grouped columns: `a`, `b`, `c`, `d`, `e`
and if `a`, `c`, and `e` are fixed, then the indexes for these will be
`1, 2, 3, 4, 5`

but FDT will render them as
`a, c, e, b, d`, since some of them are fixed, resulting in the order
`1, 3, 5, 2, 4`

So here first view port index (based on scrollable columns) will be 2
The next column will 3, but it's a fixed column, and our calculations fail.

To work around this, in groupColumns selector, I grouped the columns and assigned indexes based on the grouping (see `columnGroupIndex`).
This means that we'll have the index
`1, 2, 3, 1, 2` for 
`a, c, e, b, d`

Now since each group is continuous and maps properly to the actual column, our problem is solved.

I'm guessing _when column virtualization API is provided, we'll need the user to specify fixed columns before hand_, because it's unreasonable to run though the entire grid and find fixed columns (which ruins the virtualization api)